calculator.rb

```diff
@@ -0,0 +1,120 @@
+# Unit 1 - Ruby Fundamentals
+# Class Project - August 7, 2018
+# Calculator
+
+# Calculator operator array
+operator_options = [
+  "add", "+", "subtract", "-", "multiply", "*", "divide", "/"
+]
+
+# Intro text to user
+puts "Welcome to the Calculator program!"
+puts "Which operator would you like to use?"
+puts "1. add (+)"
+puts "2. subtract (-)"
+puts "3. multiply (*)"
+puts "4. divide (/)"
+# puts "5. modulo (%)"
+puts "Please enter the name or symbol of the desired operator."
+
+# Gets user input for operator and numbers
+print "Operator: "
+input = gets.chomp
+
+# Reprompt user until input is one of operators in operator array
+until operator_options.include? input
+  puts "Please input the name or symbol of a valid operator."
+  input = gets.chomp
+end
+
+operator = input
+
+# Gets user input for numbers to calculate
+number_reprompt = "Please enter only numeric characters. "
+
+# def get_valid_number(number_variable, ordinal)
+#   print "#{ordinal} number: "
+#   number_variable = gets.chomp
+#   until number_variable =~ /\d/
+#     puts number_reprompt
+#     print "#{ordinal} number: "
+#     number_variable = gets.chomp
+#   end
+
+print "First number: "
+first_num = gets.chomp
+  until first_num =~ /\d/
+    puts number_reprompt
+    print "First number: "
+    first_num = gets.chomp
+  end
+first_num = first_num.to_f
+
+print "Second number: "
+second_num = gets.chomp
+  until second_num =~ /\d/
+    puts number_reprompt
```

Comment:
You've written almost exactly the same code here twice, to get the first number and the second number. Could you DRY that up by putting this logic in a method?